app/controllers/pdl.py

```diff
@@ -895,13 +895,13 @@ def generate_allprograms_response(start_year, end_year):
                 formatted_result[f"{year} All Programs Total"] = value
                 total_row[f"{year} All Programs Total"] += value
 
-            # Add the final total for all programs between the years
-            formatted_result["18-22 All Programs Total"] = result_dict.get("18-22 All Programs Total", 0)
-            total_row["18-22 All Programs Total"] += formatted_result["18-22 All Programs Total"]
-
             # Append the formatted result to the state data
             state_data.append(formatted_result)
 
+        # Add the final total for all programs between the years
+        for year in range(start_year, end_year + 1):
+            total_row["18-22 All Programs Total"] += total_row[f"{year} All Programs Total"]
```

Comment:
Can we use the start and end year to make the key so it's more dynamic? I'm ok if we want to open an issue for this, but we should make it dynamic since at some point we'll want to expose the start/end years so someone could generate different ranges of data.

Reply:
Yes, we can change this to 2018-2022, but it would be better to create a new issue. I will open one.

Reply:
Yes, that makes perfect sense. I will update the PR.

Reply:
You can make it part of the issue you made, if you want. It's not urgent, but just a suggestion since don't need to hard code that key.